amlb/benchmark.py

```diff
@@ -261,10 +264,15 @@ def _append(self, board):
         Scoreboard.all(rconfig().output_dir).append(board).save()
 
     def _setup_done(self, mark=False):
-        marker_file = os.path.join(self._framework_dir, '.marker_setup_safe_to_delete')
-        setup_done = os.path.isfile(marker_file)
+        installed = os.path.join(self._framework_dir, '.installed')
```

Comment:
true, don't know why I implemented it this way originally.

Reply:
although after splitting this in to 2 methods, we see that no approch is perfect.
- single method: 2 different semantic for one method -> not nice.
- 2 methods: share common knowledge about external world (marker file) -> hidden dependency..

Reply:
To avoid your problem about shared knowledge, you can specify the marker file through a parameter (`marker_file`) which is passed on from the `setup` function which calls both. Then the knowledge of where marker file name is still contained in a single place.

Considering we would later want to expand it to allow e.g. multiple environments (for multiple framework versions) one method won't work either way.

Reply:
yes, I know that the file could be externalized... but I already regret the previous implementation tbh... what I liked with the original implementation is that everything was encapsulated in one place and avoided this logic scattering.
I'll concede though...

> Considering we would later want to expand it to allow e.g. multiple environments (for multiple framework versions) one method won't work either way.

I don't understand what you have in mind here. If you mean having multiple venv locally to run multiple versions of the framework at the same time:
1. I don't think it's very useful or a very desirable feature.
2. It means that venvs would stack up locally: not nice.
3. a single file can still hold multiple versions, i don't see the problem here.
4. as soon as we don't have any clear idea about a future change, an implementation should not be designed in details with this potential feature in mind: this is a common mistake.

Reply:
Ah, I interpreted the known issue "each framework has only one venv for all versions." as an implicit plan to support multiple venvs (for different versions). It could be a useful feature when evaluating version differences (either as a regression test or when issues arise with a new version).

> It means that venvs would stack up locally: not nice.

Agreed.